hydra-test-utils/test/Test/HSpec/MarkdownFormatterSpec.hs

```diff
@@ -0,0 +1,95 @@
+module Test.HSpec.MarkdownFormatterSpec where
+
+import Data.List (isInfixOf)
+import Hydra.Prelude
+import System.Directory (createDirectoryIfMissing)
+import System.FilePath (splitFileName, (</>))
+import Test.Hspec.Core.Format (Event (..), Format, FormatConfig)
+import Test.Hspec.Core.Runner (
+  Config (..),
+  Summary (..),
+  defaultConfig,
+  hspecWith,
+  hspecWithResult,
+ )
+import Test.Hspec.MarkdownFormatter
+import Test.Hydra.Prelude
+import Test.QuickCheck (Positive (..), Small (..), counterexample, forAll, frequency, property, vectorOf)
+import Test.QuickCheck.Monadic (assert, forAllM, monadic, monadicIO, monitor, run)
+
+spec :: Spec
+spec =
+  around (withTempDir "foo") $ do
+    it "format markdown content as expected" $ \tmpDir -> do
+      hspecWithMarkdown tmpDir testSpec
+        `shouldReturn` "# Some Spec\
+                       \\n\n\
+                       \## Sub Spec\
+                       \\n\n\
+                       \* does one thing\n\
+                       \* does two things"
+
+    it "generates markdown content to file when running spec" $ \tmpDir ->
```

Comment:
Could: be more precise what we test, i.e. "generates markdown containing all spec labels"